### ISSUE TITLE
RScript.exe does not exit

### ISSUE THREAD

**Issue by zekrom-vale** [zekrom-vale/electron-quick-start#2]
When the application exits RScript.exe does not end and has to be killed manually.
![image](https://user-images.githubusercontent.com/27376918/140544382-72c5a86f-e53e-41c2-a5d4-4c9e00cef945.png)

**Comment by zekrom-vale** [comment]
This will fix the issue for now, but does anyone know a better fix?

**Comment by zekrom-vale** [comment]
Current implementation does not allow for R to clean up and exit properly.  I could have R to listen for the page to close, clean up, and exit.  The only issue is that `onSessionEnded()` is also called on reload.  One option is just to create a new R server.

```r
server = function(input, output, session) {
  onSessionEnded(function(){ # Also is called on reload
    cleanUpFunction() # May need to await for it in some cases
    quit(save = "no")
  })
}
```

**Comment by zekrom-vale** [comment]
Allowing main.js to restart R on close resolves this issue as if R closes on `onSessionEnded` and the app is quit it just quits.  If the page is reloaded it starts a new session after R is closed.
[Added functionally to restart R if set to do so](1e86b415b83ebc020a23a4fde9e4706d1c3527c2) (Fix errors and R reload works)[314b90f3167c520b20a6daab1db258dd322b0767]

**Comment by zekrom-vale** [comment]
If the user wants to kill R and do it another way, feel free to add ideas.  As of now this solution is enough for my needs.  I do not plan on elaborating on this unless I think of a good way to do this.
